pkg/config/remote/client.go

```diff
@@ -251,20 +251,29 @@ func (c *Client) startFn() {
 // pollLoop should never be called manually and only be called via the client's `sync.Once`
 // structure in startFn.
 func (c *Client) pollLoop() {
+	firstRun := true
 	for {
 		interval := c.backoffPolicy.GetBackoffDuration(c.backoffErrorCount)
 		select {
 		case <-c.ctx.Done():
 			return
 		case <-time.After(c.pollInterval + interval):
-			c.lastUpdateError = c.update()
-			if c.lastUpdateError != nil {
-				c.backoffPolicy.IncError(c.backoffErrorCount)
-				log.Errorf("could not update remote-config state: %v", c.lastUpdateError)
+			err := c.update()
+			if err != nil {
+				if firstRun {
+					// As some clients may start before the core-agent server is up, we log the first error
+					// as a debug log as the race is expected. If the error persists, we log with error logs
+					log.Debugf("retrying update of remote-config state after cold start (%v)", err)
+				} else {
+					c.lastUpdateError = err
```

Comment:
Compared to previous code, `c.lastUpdateError` will always contain the last error instead of being set to `nil` if last update was successful.

And if I understand well: it means as soon as an error occured, `newUpdateRequest()` will always report there is an error. Is it expected ?

Reply:
It's not, good catch